netadapter/server/grpcserver/grpc_connection.go

```diff
@@ -83,8 +84,13 @@ func (c *gRPCConnection) Disconnect() {
 	close(c.stopChan)
 
 	if c.isOutbound {
-		clientStream := c.stream.(protowire.P2P_MessageStreamClient)
-		_ = clientStream.CloseSend() // ignore error because we don't really know what's the status of the connection
+		spawn("gRPCConnection.Disconnect-clientStream.CloseSend", func() {
+			// Wait a second before closing the stream, to let the send queue to get emptied.
+			const finishSendDuration = time.Second
+			time.Sleep(finishSendDuration)
+			clientStream := c.stream.(protowire.P2P_MessageStreamClient)
+			_ = clientStream.CloseSend() // ignore error because we don't really know what's the status of the connection
+		})
 	}
 
 	log.Debugf("Disconnected from %s", c)
```

Comment:
Please move this log into the goroutine above and consider adding a "Disconnecting..." log.